scipy/interpolate/_interpolate.py

```diff
@@ -92,11 +92,28 @@ def lagrange(x, w):
 # !! found, get rid of it!
 
 
+dep_mesg = """\
+`interp2d` is deprecated in SciPy 1.10 and will be removed in a future SciPy version. 
```

Comment:
I think it is better if we commit to a version. As a user I means I know when I have to change my code by and for maintainers it makes it more obvious when to execute the deprecation.
```suggestion
`interp2d` is deprecated in SciPy 1.10 and will be removed in SciPy 1.12.0. 
```